src/app/components/annotator/text-annotator/text-annotator.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { AnnotatorService } from 'src/app/services/annotator/annotator.service';
+
+@Component({
+  selector: 'evt-text-annotator',
+  templateUrl: './text-annotator.component.html',
+  styleUrls: ['./text-annotator.component.scss'],
+})
+export class TextAnnotatorComponent implements OnInit {
+
+  public selectedText: string;
+  public showAdder = false;
+  public menuPosition = {menu: {x: 0, y: 0}};
+
+  constructor(
+    private annotator: AnnotatorService,
+  ) {}
+
+  ngOnInit(): void {
+      this.annotator.textSelection$.subscribe((selection) => {
+      this.selectedText = selection.toString();
+      /\S/.test(selection.toString())
```

Comment:
It is better to use an `if` statement instead of the ternary operator.
Side effects inside expression evaluation should be avoided